### ISSUE TITLE
[Bug]: Getting "cannot be used as a JSX component" typescript error after upgrading to React 19 

### ISSUE THREAD

**Issue by skaionedev** [react-icons/react-icons#1006]
### Describe the bug

 Getting "cannot be used as a JSX component" typescript error after upgrading to React 19 
 
   `"ICON_NAME_HERE" cannot be used as a JSX component.
 Its type 'IconType' is not a valid JSX element type.
    Type 'IconType' is not assignable to type '(props: any) => ReactNode'.
      Type 'Element' is not assignable to type 'ReactNode'.
        Property 'children' is missing in type 'ReactElement<any, any>' but required in type 'ReactPortal'.`

### Expected behavior

After changing types from `export type IconType = (props: IconBaseProps) => JSX.Element;` to `export type IconType = (props: IconBaseProps) => React.ReactNode;` issue is resolved. 
Please update types to make it React 19 compatible

### Minimal sample repository URL

-

### Steps To Reproduce

1. Create React 19 + Vite + Typescript app or update to 19 version
2. Create component with any icon from react-icons package

`import { RiBarChartFill } from 'react-icons/ri'`

```
export const Example = () => {
  return (
    <div>
      <RiBarChartFill />
    </div>
  )
}
```

### Package version

^5.4.0

### Relevant log output

```Shell
'RiBarChartGroupedFill' cannot be used as a JSX component.
  Its type 'IconType' is not a valid JSX element type.
    Type 'IconType' is not assignable to type '(props: any) => ReactNode'.
      Type 'Element' is not assignable to type 'ReactNode'.
        Property 'children' is missing in type 'ReactElement<any, any>' but required in type 'ReactPortal'.ts(2786)
index.d.ts(387, 9): 'children' is declared here.
```

**Comment by prabhjotsinghchadha** [comment]
I am getting same error

**Comment by SallyHabib** [comment]
same bug here

**Comment by pepew-le-boss** [comment]
same

**Comment by Bjodol** [comment]
There is a PR open to resolve this: https://github.com/react-icons/react-icons/pull/1004

**Comment by diego-aquino** [comment]
I managed to work around this issue in `react-icons@5.4.0` by declaring a custom `IconType` compatible with React 19:

`react-icons.d.ts`

```ts
import { JSX } from 'react'
import { IconBaseProps } from 'react-icons'

declare module 'react-icons/lib/index' {
  export type IconType = (props: IconBaseProps) => JSX.Element
}
```

**Comment by dvrfluxchat** [comment]
@diego-aquino did you do any changes to tsconfig as well?

**Comment by diego-aquino** [comment]
> [@diego-aquino](https://github.com/diego-aquino?rgh-link-date=2025-01-24T12%3A29%3A27.000Z) did you do any changes to tsconfig as well?

@dvrfluxchat, no, I didn't need to. I just created the file `react-icons.d.ts` (in `src/types` in my case).

**Comment by ohabash** [comment]
`react-icons.d.ts` in _root_ or in `src/types` did not work. Is there something else i can do? Or am i installing the work around wrong?

```
System:
    OS: macOS 15.1.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 119.27 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
  Browsers:
    Chrome: 132.0.6834.160
    Safari: 18.1.1
  npmPackages:
    next: 14.2.16 => 14.2.16 
    react: 18.2.0 => 18.2.0 
    react-icons: ^5.4.0 => 5.4.0 
```

**Comment by ohabash** [comment]
UPDATE

had to make sure i installed these packages  (for me was 18)

`npm i @types/react@18 @types/react-dom@18`

**Comment by Ikechukwu-attah** [comment]
npm i @types/react@18 @types/react-dom@18
This solves the problem for me.

**Comment by irios-phorus** [comment]
Installing @types/react@18 @types/react-dom@18 when you are using react 19 only creates compatibility issues. It shouldn't be the recommended solution and doesn't even serve as a workaround in many cases, as it can break your project.
The library shouldn't be using deprecated stuff in the first place

**Comment by irios-phorus** [comment]
For now, just re-declare the JSX namespace yourselves.

Create a `src/types/jsx.d.ts` file with:
```ts
declare namespace JSX {
  export import IntrinsicElements = React.JSX.IntrinsicElements
  export import Element = React.JSX.Element
}
```

And make sure tsconfig.json includes this file:
```json
  "include": ["src", "src/types/jsx.d.ts"]
```

**Comment by valento** [comment]
I applied the configuration for NextJS-app (but still on create-react-app boiler), as suggested [here](https://www.chakra-ui.com/docs/get-started/frameworks/next-app), and it worked for me:
```
// tsconfig.json
{
  "compilerOptions": {...
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "Bundler",
  }
}
```
I hope it works for you too

**Comment by MrScotch679** [comment]
As an option you can downgrade to 4.10.1 version, it works for me

**Comment by varguitas** [comment]
> I applied the configuration for NextJS-app (but still on create-react-app boiler), as suggested [here](https://www.chakra-ui.com/docs/get-started/frameworks/next-app), and it worked for me:
> 
> ```
> // tsconfig.json
> {
>   "compilerOptions": {...
>     "target": "ESNext",
>     "module": "ESNext",
>     "moduleResolution": "Bundler",
>   }
> }
> ```
> 
> I hope it works for you too

This one worked for me! Thanks!

**Comment by appsparkler** [comment]
This is what worked for me:

```tsx
import { SomeIcon } from "react-icons/fa6";

const SomeIconRef= SomeIcon as React.ElementType

export const Navbar = () => {
    return (
        <div className="navbar bg-base-100 shadow-sm">
            <div className="navbar-end">
                <a className="btn">
                    <SomeIconRef />
                </a>
            </div>
        </div>
    )
}
```

If there are many icons that are re-used throughout the application; we can create an icons.ts file and export from there:

```ts
import { IconOne } from 'react-icons/fa'
import { IconTwo } from 'react-icons/fa'

export const IconOneRef = IconOne as React.ElementType
export const IconTwoRef = IconTwo as React.ElementType
```

Now we can import straightaway from `icon.ts`

```tsx
import {IconTwoRef} from './path/to/icon.ts';

// IconTwoRef works fine :)
```

Good Luck!

**Comment by stivenm16** [comment]
> As an option you can downgrade to 4.10.1 version, it works for me

This worked to me too, thanks!

**Comment by ohabash** [comment]
> Installing @types/react@18 @types/react-dom@18 when you are using react 19 only creates compatibility issues. It shouldn't be the recommended solution and doesn't even serve as a workaround in many cases, as it can break your project. The library shouldn't be using deprecated stuff in the first place

dont install the wrong version 💡

**Comment by Astralzz** [comment]
I solved this by reverting to version **5.0.3** and removing the `^`.

**Before:**
```json
"react-icons": "^5.3.0"
```

**After:**
```json
"react-icons": "5.3.0"
```

**Comment by suttonb** [comment]
@skaionedev Could you provide more info on the closed status please? I.e. is it one of the following:

1. One of the workarounds posted in the comments above is considered the accepted solution? (If so, which one?)
2. A PR with a fix has been merged, but not included in a release version yet?
3. A fix has been included in a new release?

Or is it something else?

**Comment by itaisir** [comment]
Use "react-icons": "5.3.0" with out the ^

and
"devDependencies": {
"@types/react": "^19.0.10",
"@types/react-dom": "^19.0.4",
"typescript": "^4.9.5"
}

100% is working 🎉🎉🎉🎉🎉

**Comment by PDK-projects** [comment]
> I managed to work around this issue in `react-icons@5.4.0` by declaring a custom `IconType` compatible with React 19:
> 
> `react-icons.d.ts`
> 
> import { ReactNode } from 'react'
> import { IconBaseProps } from 'react-icons'
> 
> declare module 'react-icons/lib/index' {
>   export type IconType = (props: IconBaseProps) => **ReactNode**
> }

Changed **ReactNode** to **ReactElement | null**
